### PR TITLE
feat: added a vesting.md documentaion

### PR DESCRIPTION
**Pull request by more-of-odeh**
Added a vesting.md markdown file that explains the vesting component for protocols looking to use Konoha.

### CODE REVIEW

File: docs/vesting.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+2. Contract Integration: The vest function should be exported externally from the contract to allow protocols to execute the vesting process.
+
+
+
+## Components
+
+The Vesting contract is composed of the following components:
+
+1. VestingImpl: Implementation of the IVesting trait for managing vesting operations.
+2. Storage: Storage component for storing vesting milestones.
```
Comment (tensojka) [inline docs/vesting.md:90]: What?? The whole vesting.cairo is technically a component. What is the point of this heading?

File: docs/vesting.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+### Events
+
+The Vesting contract emits the following events:
+
+1. VestingMilestoneAdded: This indicates the addition of a new vesting milestone.
+2. Vested: This indicates that tokens have been vested for a grantee.
+
+## Additional Customizations
+
+For additional customization possibilities such as options on the governance token, further modifications can be explored within the contract.
```
Comment (tensojka) [inline docs/vesting.md:101]: Is this written by chatgpt? Makes no sense.

**Review by tensojka — CHANGES_REQUESTED** [review]
Thanks for submitting what I assume is a work in progress. You can use the Draft pull request functionality in Github to mark that it's not yet ready for review. But thanks for keeping me up to date on your work.

What is lacking is among others an example of how exactly the Linear vesting schedule works. Imagine that you are a protocol that wants to implement Vesting of their token for their team and they open documentation. Will they understand what Konoha can offer and what it cannot? And how easy it is to integrate it with whatever solution they currently use? Currently, it would not be clear. Please rework.

File: docs/vesting.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+1. vest(grantee: ContractAddress, vested_timestamp: u64): This function executes the vesting process for a specific grantee at a given timestamp.
+2. add_vesting_milestone(vesting_timestamp: u64, grantee: ContractAddress, amount: u128): This function adds a vesting milestone for a grantee at a specific vesting timestamp.
+3. add_linear_vesting_schedule(first_vest: u64, period: u64, increments_count: u64, total_amount: u128): This function adds a linear vesting schedule with specified parameters.
+
+
+
+###  Usage
+
+# Executing Vesting
+To execute the vesting process for a grantee at a specified timestamp, use the vest function. This function will check if the conditions are met and, if so, mint the vested tokens to the grantee.
```
[the remark this answers is not in the document]
Comment (tensojka) [reply docs/vesting.md]: > > To vest a schedule, you need to call vest() multiple times with the exactly correct times. Best way to find this is to read the VestingMilestoneAdded event, this is not documented anywhere.
> 
> @tensojka I think this is done on the `Linear vesting schedule`. I stand to be corrected.

I'm sorry, your message is unclear. The event is indeed documented, but at the end. What isn't is – I have set up a Linear vesting schedule, how do I vest it over time?

**Comment by tensojka** [comment]
> Imagine that you are a protocol that wants to implement Vesting of their token for their team and they open documentation. Will they understand what Konoha can offer and what it cannot?

you fixed the few concere issues I raised but didn't address the bigger picture.

**Comment by tensojka** [comment]
once ready for review, please re-request review

File: docs/vesting.md [hunk 4]
```diff
@@ -38,10 +48,13 @@ To execute the vesting process for a grantee at a specified timestamp, use the v
 
 #### Example usage of vesting tokens
 
+```
```
Comment (tensojka) [inline docs/vesting.md:51]: Missing information on how the cliff works

File: docs/vesting.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+- Storage: Storage component for storing vesting milestones.
+
+### Events
+
+The Vesting contract emits the following events:
+
+- VestingMilestoneAdded: This indicates the addition of a new vesting milestone.
+- Vested: This indicates that tokens have been vested for a grantee.
+
+Note that the methods `add_vesting_milestone ` and `add_linear_vesting_schedule` can only be called by the contract itself to prevent unauthorized modifications.
```
Comment (tensojka) [inline docs/vesting.md:119]: Again! What question does this _answer_? How should this be used with Konoha? How can the contract call itself?

Please refer to my first review. You fixed the few concrete issues I raised, but did not address the bigger picture.

File: docs/vesting.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+To customize the Vesting contract for specific protocol’s need, use the methods defined below;
+
+- Custom Proposals: To add new vesting schedules, the component should be part of a custom proposal that calls either add_linear_vesting_schedule or add_vesting_milestone.
+- Contract Integration: The vest function should be exported externally from the contract to allow protocols to execute the vesting process.
+
+The Vesting contract is composed of the following:
+
+- VestingImpl: Implementation of the IVesting trait for managing vesting operations.
+- Storage: Storage component for storing vesting milestones.
```
Comment (tensojka) [inline docs/vesting.md:110]: Why is this even here? Again –– what question does this answer? Who cares?

**Review by tensojka — CHANGES_REQUESTED** [review]
Please refer to my first review and rework it. I would suggest you rewrite from scratch.

e.g. you do not say what is a timestamp.

I would also suggest you try to set up vesting on your konoha instance and write down everything you needed to do that – that will be good documentation, not this, which could have been as well been written by a LLM...

File: docs/vesting.md (hunk 5 above)
Comment (more-of-odeh) [reply docs/vesting.md:119]: It's answers the question of the current limitations to it.

> Again! What question does this _answer_? How should this be used with Konoha? How can the contract call itself?
>

Comment (more-of-odeh) [reply docs/vesting.md:119]: This will help to avoid Self-call Restrictions.

**Comment by tensojka** [comment]
Again, you only addressed the concrete minor issues I raised but not the bigger picture. Closing

**Pull request closed without merging**